plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/DeltaLakeTableMetadataScheduler.java

```diff
@@ -0,0 +1,230 @@
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        if (enabled) {
+            int threads = storeTableMetadataThreads.orElse(0);
+            executor = threads == 0 ? newDirectExecutorService() : newFixedThreadPool(threads, threadsNamed("store-table-metadata-%s"));
+            scheduler = newSingleThreadScheduledExecutor(daemonThreadsNamed("store-table-metadata"));
+            scheduler.scheduleWithFixedDelay(this::process, 0, storeTableMetadataInterval.toMillis(), MILLISECONDS);
+        }
+    }
+
+    @VisibleForTesting
+    public void process()
+    {
+        Map<SchemaTableName, UpdateInfo> updateTables;
+        synchronized (this) {
+            updateTables = updateInfos.entrySet().stream()
+                    .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue, maxBy(comparing(UpdateInfo::version))));
+            updateInfos.clear();
+        }
+
+        log.debug("Processing %s table(s): %s", updateTables.size(), updateTables.keySet());
+        for (Map.Entry<SchemaTableName, UpdateInfo> entry : updateTables.entrySet()) {
+            executor.execute(() -> updateTable(entry.getKey(), entry.getValue()));
+        }
+    }
+
+    private void updateTable(SchemaTableName schemaTableName, UpdateInfo info)
+    {
+        log.debug("Updating table: '%s'", schemaTableName);
+        HiveMetastore metastore = hiveMetastoreFactory.createMetastore(Optional.of(info.identity));
+        try {
+            metastore.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName())
+                    .ifPresent(table -> {
+                        Table updatedTable = setTableMetadata(table, info.version, info.schemaString, info.tableComment);
+                        metastore.replaceTable(table.getDatabaseName(), table.getTableName(), updatedTable, buildInitialPrivilegeSet(table.getOwner().orElseThrow()));
+                        log.debug("Replaced table: '%s'", schemaTableName);
+                    });
+        }
+        catch (Exception e) {
+            log.warn(e, "Failed to store table metadata for '%s'", schemaTableName);
+        }
+    }
+
+    @VisibleForTesting
+    public void clear()
+    {
+        updateInfos.clear();
+    }
+
+    @PreDestroy
+    public void stop()
+    {
+        if (enabled) {
+            scheduler.shutdownNow();
+            executor.shutdownNow();
+        }
+    }
+
+    public boolean isSameTransactionVersion(Table table, TableSnapshot snapshot)
+    {
+        if (!containsLastTransactionVersion(table)) {
+            return false;
+        }
+        return getLastTransactionVersion(table) == snapshot.getVersion();
+    }
+
+    public boolean containsLastTransactionVersion(Table table)
+    {
+        return table.getParameters().containsKey(TRINO_LAST_TRANSACTION_VERSION);
+    }
+
+    public long getLastTransactionVersion(Table table)
+    {
+        return Long.parseLong(table.getParameters().get(TRINO_LAST_TRANSACTION_VERSION));
+    }
```

Comment:
- you can squash these methods into one, returning optional
same for others.

- should they be static?